### PR TITLE
RANGER-3985: use table creation rule for trino

### PR DESCRIPTION
**Pull request by jonashartwig**

### CODE REVIEW

File: plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java [hunk 1]
```diff
@@ -132,7 +132,14 @@ public void testTableOperations()
...
 
-    accessControlManager.checkCanCreateTable(context(alice), aliceTable,Map.of());
+    accessControlManager.checkCanCreateTable(context(alice), aliceTable, Map.of());
+    assertThrows(AccessDeniedException.class, () -> {
+      accessControlManager.checkCanCreateTable(
+        context(alice),
+        new CatalogSchemaTableName("alice-catalog", "schema","wrong-table"),
+        Map.of()
+      );
+    });
```
Comment (emartgu) [inline plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java]: Since they use guava ImmutableSet, maybe we should use guava ImmutableMap (which also needs to be imported). I don't know if they are picky around this.

```suggestion
    accessControlManager.checkCanCreateTable(context(alice), aliceTable, ImmutableMap.of());
    assertThrows(AccessDeniedException.class, () -> {
      accessControlManager.checkCanCreateTable(
        context(alice),
        new CatalogSchemaTableName("alice-catalog", "schema", "wrong-table"),
        ImmutableMap.of()
      );
    });
```

File: pom.xml [hunk 2]
```diff
@@ -330,6 +330,11 @@
             <activation>
                <jdk>11</jdk>
             </activation>
+            <properties>
+                <java.version.required>11</java.version.required>
+                <javac.source.version>11</javac.source.version>
+                <javac.target.version>11</javac.target.version>
+            </properties>
```
Comment (emartgu) [inline pom.xml]: Maybe this should go to a separate PR.

Comment (jonashartwig) [reply pom.xml]: maybe, we will see what (if) they say anything, without this it does not compile

File: plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java (hunk 1 above)
Comment (jonashartwig) [reply plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java]: I copied that from exisiting tests. I rather not change this.

File: pom.xml (hunk 2 above)
Comment (jonashartwig) [reply pom.xml]: you are right, somehow now it builds without, i remove it

File: plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java (hunk 1 above)
Comment (emartgu) [reply plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java]: Ok, can you at least add a space before "wrong-table" :)

File: plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java [hunk 3]
```diff
@@ -132,7 +132,14 @@ public void testTableOperations()
     assertEquals(accessControlManager.filterTables(context(alice), aliceCatalog, aliceTables), aliceTables);
     assertEquals(accessControlManager.filterTables(context(bob), "alice-catalog", aliceTables), ImmutableSet.of());
 
-    accessControlManager.checkCanCreateTable(context(alice), aliceTable,Map.of());
+    accessControlManager.checkCanCreateTable(context(alice), aliceTable, Map.of());
+    assertThrows(AccessDeniedException.class, () -> {
+      accessControlManager.checkCanCreateTable(
+        context(alice),
+        new CatalogSchemaTableName("alice-catalog", "schema","wrong-table"),
```
Comment (emartgu) [inline plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java]:
```suggestion
        new CatalogSchemaTableName("alice-catalog", "schema", "wrong-table"),
```

Comment (jonashartwig) [reply plugin-trino/src/test/java/org/apache/ranger/authorization/trino/authorizer/RangerSystemAccessControlTest.java]: done

**Review by emartgu — APPROVED** [review]

**Review by tarekabouzeid — APPROVED** [review]
